### PR TITLE
Add association variables to add and edit templates

### PR DESCRIPTION
**Pull request by mentisy**
Adds proper docblock variables for associations in `add` and `edit`templates.

See #759

### CODE REVIEW

**Comment by dereuromark** [comment]
I think IdeHelper does this already afterwards. But doesnt hurt to generate it right away 
👍

**Comment by markstory** [comment]
Thank you 👍

File: templates/bake/Template/add.twig [hunk 1]
```diff
@@ -17,6 +17,19 @@
 /**
  * @var \{{ namespace }}\View\AppView $this
  * @var \{{ entityClass }} ${{ singularVar }}
+        {{- "\n" }}
+{%- if associations.BelongsTo is defined %}
+    {%- for assocName, assocData in associations.BelongsTo %}
+ * @var array ${{ assocData.variable }}
```
Comment (ADmad) [inline templates/bake/Template/add.twig:23]: Lies!! This var is a result of `find('list')` which is a `Query` instance, not an array. So the type should be `iterable`.

Comment (dereuromark) [reply templates/bake/Template/add.twig:23]: The concrete type isnt that important usually just the existence:)
Could usually also be string[] etc
As those are used for options.

Comment (dereuromark) [reply templates/bake/Template/add.twig:23]: The concrete type isnt that important usually just the existence:)
Could usually also be string[] etc
As those are used for options.

Comment (ADmad) [reply templates/bake/Template/add.twig:23]: > The concrete type isnt that important...

I disagree, incorrect types are misleading and cause confusion for new users.

Comment (mentisy) [reply templates/bake/Template/add.twig:23]: @ADmad : I was thinking about this when adding the variable. On the index templates, you have this docblock:
```php
/**
 * @var \App\View\AppView $this
 * @var \App\Model\Entity\File[]|\Cake\Collection\CollectionInterface $files
 */
```

We could do something similar with add/edit.

```php
/**
 * @var \App\View\AppView $this
 * @var string[]|\Cake\Collection\CollectionInterface $files
 */
```

What do you think? I can create a PR for this if you'd like.

Comment (ADmad) [reply templates/bake/Template/add.twig:23]: `\Cake\Collection\CollectionInterface<string>` would be the most accurate but if that's not supported by IDEs(PhpStorm) yet, then `string[]|\Cake\Collection\CollectionInterface` is fine.

Comment (othercorey) [reply templates/bake/Template/add.twig:23]: Please use `\Cake\Collection\CollectionInterface|array<string>` or `\Cake\Collection\CollectionInterface|string[]` to follow our new standard.

Comment (mentisy) [reply templates/bake/Template/add.twig:23]: Even if it was, wouldn't it be a good idea to follow the conventions in other templates until all templates are changed?

I couldn't get it to work with PhpStorm 2021.1.4 at least.

While `\Cake\Collection\CollectionInterface<string>` doesn't throw any errors/warnings, say we tried `\Cake\Collection\CollectionInterface<\Cake\Datasource\EntityInterface>` for index it wouldn't suggest any properties/methods in `EntityInterface` when looping through it. So it doesn't seem to be fully supported yet.

I say we go with the `string[]|\Cake\Collection\CollectionInterface` way.

**Pull request merged**